src/gadgets/mpt_update.rs

```diff
@@ -560,39 +1844,255 @@ mod test {
 
     #[derive(Clone, Debug)]
     struct TestCircuit {
-        updates: Vec<SMTTrace>,
+        proofs: Vec<Proof>,
     }
 
     impl TestCircuit {
+        fn new(traces: Vec<(MPTProofType, SMTTrace)>) -> Self {
+            Self {
+                proofs: traces.into_iter().map(Proof::from).collect(),
+            }
+        }
+
         fn hash_traces(&self) -> Vec<(Fr, Fr, Fr)> {
-            vec![]
+            let mut hash_traces = vec![(Fr::zero(), Fr::zero(), Fr::zero())];
```

Comment:
@noel2004 `hash_traces` generate the poseidon lookup triplet (input0, input1, result).   Do you think it is needed to put (Fr::zero(), Fr::zero(), Fr::zero()) here first?

Reply:
I think this triplet should not be used in the refactored code since it is just a trick in the old code. This triplet has brought a insecure feature by enabling a non-existed hash (0, 0) -> 0 and maybe harmful in storage value.

Reply:
yes, that's a good point. it should be possible to change the constraint builder to lookup hash(1, 1) = the correct value when the condition evaluates to false.